client/my-sites/domains/components/form/hidden-input.jsx

```diff
@@ -3,56 +3,53 @@
  *
  * @format
  */
-
-import { isEmpty } from 'lodash';
-import React from 'react';
+import React, { PureComponent } from 'react';
+import isEmpty from 'lodash/isEmpty';
 
 /**
  * Internal dependencies
  */
 import Input from './input';
 
-export default React.createClass( {
-	displayName: 'HiddenInput',
-
-	componentWillReceiveProps: function( nextProps ) {
-		if ( ! this.state.toggled && ! isEmpty( nextProps.value ) ) {
-			this.setState( { toggled: true } );
-		}
-	},
-
-	getInitialState: function() {
-		return {
-			toggled: false,
+export class HiddenInput extends PureComponent {
+	constructor( props, context ) {
+		super( props, context );
+		this.state = {
+			toggled: ! isEmpty( props.value ),
 		};
-	},
+		this.inputField = null;
+	}
 
-	componentDidUpdate: function( prevProps, prevState ) {
-		// Focus the input only when the user explicitly clicked the toggle link
-		if ( ! prevState.toggled && this.state.toggled && prevProps.value === this.props.value ) {
-			this.refs.input.focus();
-		}
-	},
-
-	handleClick: function( event ) {
+	handleClick = event => {
 		event.preventDefault();
 
-		this.setState( {
-			toggled: true,
-		} );
-	},
+		this.setState(
+			{
+				toggled: true,
+			},
+			() => {
+				this.inputField && this.inputField.focus();
+			}
+		);
+	};
+
+	assignInputFieldRef = input => {
+		this.inputField = input;
+	};
 
-	render: function() {
+	render() {
 		if ( this.state.toggled ) {
-			return <Input ref="input" { ...this.props } />;
+			return <Input ref={ this.assignInputFieldRef } { ...this.props } />;
 		}
 
 		return (
-			<div className="hidden-input">
+			<div className="form__hidden-input hidden-input">
```

Comment:
Could you also rename [the `hidden-input` class](https://github.com/Automattic/wp-calypso/blob/aab6aa0f85d32f6af85f2bf44bd2224d5b23ba13/client/my-sites/checkout/checkout/style.scss#L598)? It doesn't make sense to keep both of them, especially since `hidden-input` is not a valid one (in terms of our CSS naming rules).

Reply:
👍 I'll get rid of `hidden-input` and update any related CSS decs.